app/workers/set_tenant_id_worker.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class SetTenantIdWorker < ApplicationJob
+
+  def perform(provider_batch)
+    provider_batch.each do |provider|
+      provider.master = false
+      provider.save!
+
+      {
+        BackendApi => :account_id,
+        LogEntry => :provider_id,
+        Alert => :account_id,
+      }.each do |model, relation|
+        q = {relation => provider.id}
+        
+        model.where(q).each do |instance|
```

Comment:
```suggestion
        model.where(relation => provider.id).find_each do |instance|
```

Prevent loading all objects in memory